src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts

```diff
@@ -20,14 +20,19 @@ interface DotEditPageNavItem {
     templateUrl: './dot-edit-page-nav.component.html',
     styleUrls: ['./dot-edit-page-nav.component.scss']
 })
-export class DotEditPageNavComponent implements OnInit {
+export class DotEditPageNavComponent implements OnChanges {
     @Input() pageState: DotRenderedPageState;
     model: Observable<DotEditPageNavItem[]>;
+    isEnterpriseLicense: boolean;
 
     constructor(private dotLicenseService: DotLicenseService, public dotMessageService: DotMessageService, public route: ActivatedRoute) {}
 
-    ngOnInit(): void {
-        this.model = this.dotMessageService
+    ngOnChanges(): void {
+        this.model = !this.model ? this.loadData() : Observable.of(this.getNavItems(this.pageState, this.isEnterpriseLicense));
```

Comment:
This looks problematic to me... I like the idea crate a class property of ```isEnterpriseLicense``` so we don't have a to do a request to get the license every time, but this could have some side effects, if the license is apply after this components loads the first time we'll never see the update. So I think we should call ```this.loadData()``` all the time, the messages are cached so that request will not happen (or at least it shouldn't)

Reply:
Also do you double check that this ```ngOnChanges``` is not getting called when we don't want to?